### PR TITLE
IQ-125 Review flow for observation/survey

### PR DESCRIPTION
**Pull request by Vivek-M-08**
@krgauraw
Design doc approved by DC - https://project-sunbird.atlassian.net/l/cp/HDuR6Nmz
Story - https://project-sunbird.atlassian.net/browse/IQ-125

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions: Java 11, scala-2.11, play-2.7.2
* Hardware versions: 2 CPU/ 4GB RAM

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by krgauraw** [comment]
@Vivek-M-08 Please write test cases to have coverage at least 80%
cc: @sajeshkayyath

**Comment by Vivek-M-08** [comment]
@krgauraw - To ensure the desired coverage, I have added required test-cases.

File: assessment-api/assessment-actors/src/main/scala/org/sunbird/managers/AssessmentManager.scala [hunk 1]
```diff
@@ -299,4 +329,24 @@ object AssessmentManager {
...
+		val fields: util.List[String] = JavaConverters.seqAsJavaListConverter(request.get("fields").asInstanceOf[String].split(",").filter(field => StringUtils.isNotBlank(field) && !StringUtils.equalsIgnoreCase(field, "null"))).asJava
+		request.getRequest.put("fields", fields)
+		val res = new java.util.ArrayList[java.util.Map[String, AnyRef]] {}
+		DataNode.read(request).map(node => {
+			val metadata = new java.util.HashMap().asInstanceOf[java.util.Map[String, AnyRef]]
+			metadata.put("identifier", node.getIdentifier.replace(".img", ""))
+			metadata.put("comment", node.getMetadata.getOrDefault("rejectComment", ""))
+			res.add(metadata)
+			if (!StringUtils.equalsIgnoreCase("QuestionSet", node.getObjectType))
+				throw new ClientException("ERR_QUESTION_SET_ADD", s"Node with Identifier ${node.getIdentifier} is not a Question Set.")
```
Comment (krgauraw) [inline assessment-api/assessment-actors/src/main/scala/org/sunbird/managers/AssessmentManager.scala:343]: Please change the error code

File: assessment-api/assessment-actors/src/main/scala/org/sunbird/v5/managers/AssessmentV5Manager.scala [hunk 2]
```diff
@@ -489,4 +520,24 @@ object AssessmentV5Manager {
...
+    val fields: util.List[String] = JavaConverters.seqAsJavaListConverter(request.get("fields").asInstanceOf[String].split(",").filter(field => StringUtils.isNotBlank(field) && !StringUtils.equalsIgnoreCase(field, "null"))).asJava
+    request.getRequest.put("fields", fields)
+    val res = new java.util.ArrayList[java.util.Map[String, AnyRef]] {}
+    DataNode.read(request).map(node => {
+      val metadata = new java.util.HashMap().asInstanceOf[java.util.Map[String, AnyRef]]
+      metadata.put("identifier", node.getIdentifier.replace(".img", ""))
+      metadata.put("comment", node.getMetadata.getOrDefault("rejectComment", ""))
+      res.add(metadata)
+      if (!StringUtils.equalsIgnoreCase("QuestionSet", node.getObjectType))
+        throw new ClientException("ERR_QUESTION_SET_ADD", s"Node with Identifier ${node.getIdentifier} is not a Question Set.")
```
Comment (krgauraw) [inline assessment-api/assessment-actors/src/main/scala/org/sunbird/v5/managers/AssessmentV5Manager.scala:533]: please change the error code

**Review by krgauraw — CHANGES_REQUESTED** [review]

**Review by sajeshkayyath — APPROVED** [review]

**Pull request merged**